base/statistics.jl

```diff
@@ -545,6 +545,9 @@ function histrange{T<:Integer,N}(v::AbstractArray{T,N}, n::Integer)
     if length(v) == 0
         return 0:1:0
     end
+    if n <= 0
+        throw(DomainError())
```

Comment:
I would tend to throw an `ArgumentError("number of bins n=$n must be positive")` with a descriptive message.  We tend to use `DomainError` for more "mathy" functions.